In [2]:
import pandas as pd 

## Dive into Data Science - Bradford Tuckfield 
### Pandas 
Pandas is an open source Python library, primarily used to analyze and manipulate data.

In [3]:
hour = pd.read_csv('hour.csv')
print(hour.head())

   instant      dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1   0        0        6           0   
1        2  2011-01-01       1   0     1   1        0        6           0   
2        3  2011-01-01       1   0     1   2        0        6           0   
3        4  2011-01-01       1   0     1   3        0        6           0   
4        5  2011-01-01       1   0     1   4        0        6           0   

   weathersit  temp   atemp   hum  windspeed  casual  registered  count  
0           1  0.24  0.2879  0.81        0.0       3          13     16  
1           1  0.22  0.2727  0.80        0.0       8          32     40  
2           1  0.22  0.2727  0.80        0.0       5          27     32  
3           1  0.24  0.2879  0.75        0.0       3          10     13  
4           1  0.24  0.2879  0.75        0.0       0           1      1  


read_csv() = Reads the file data <br>
pd.head() = Returns the first N rows (5 by default).

In [6]:
print(f"Mean of Total Count: {hour['count'].mean()}")
print(f"Median of Total Count: {hour['count'].median()}")
print(f"Maximum Registered : {hour['registered'].max()}")
print(f"Minimum Registered : {hour['registered'].min()}")
print(f"Standard Deviation of Total Count: {hour['count'].std()}")

Mean of Total Count: 189.46308763450142
Median of Total Count: 142.0
Maximum Registered : 886
Minimum Registered : 0
Standard Deviation of Total Count: 181.38759909186476


## Summary Statistics 
### Simple calculations which help us understand the data closely and understand the nature of it. All the functions are performed through pandas.
Mean - Sum of all values divided by total no. of values (Average)   .mean() <br>
Median - The middle value of a series (In case of even numbers, the middle two are added and divided by two)  .median() <br>
Max - Maximum Value within the range .max() <br>
Min - Minimum value within the range .min() <br>
Standard Deviation - Shows how far is the data from its mean value. Ex : +- 181  .std() <br>
